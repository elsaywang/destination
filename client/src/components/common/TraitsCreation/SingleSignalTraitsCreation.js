import React from 'react';
import PropTypes from 'prop-types';
import styles from './TraitsCreation.css';
import Add from '@react/react-spectrum/Icon/Add';

const SingleSignalTraitsCreation = ({
    createTraitUrl,
    traitsCreationLabelText,
    canCreateTrait,
}) => {
    if (canCreateTrait) {
        return (
            <a className="spectrum-Link" href={createTraitUrl}>
                <div className={styles.singleCreation}>
                    <Add size="S" />
                    <div className={styles.singleCreationlinkText}>{traitsCreationLabelText}</div>
                </div>
            </a>
        );
    }

    return '0 Traits';
};

SingleSignalTraitsCreation.propTypes = {
    createTraitUrl: PropTypes.string,
    traitsCreationLabelText: PropTypes.string,
    canCreateTrait: PropTypes.bool,
};

export default SingleSignalTraitsCreation;
