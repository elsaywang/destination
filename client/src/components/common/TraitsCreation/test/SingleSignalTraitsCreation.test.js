import React from 'react';
import { shallow } from 'enzyme';
import Add from '@react/react-spectrum/Icon/Add';
import SingleSignalTraitsCreation from '../SingleSignalTraitsCreation';

describe('<SingleSignalTraitsCreation/> component', () => {
    const mockFn = jest.fn();
    const props = {
        traitsCreationLabelText: 'Create Rule-Based Trait',
        canCreateTraits: true,
        storeSessionAndNavigateToTraits: mockFn,
    };

    describe('when user has trait creation permission', () => {
        const wrapper = shallow(<SingleSignalTraitsCreation {...props} />);

        it('matches the snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        });

        it('includes <Add/> icon with size=`S`', () => {
            expect(wrapper.find(Add).exists()).toBe(true);
            expect(wrapper.find(Add).props().size).toEqual('XS');
        });

        it('renders the correct text passed from `traitsCreationLabelText` prop in a <div/>', () => {
            expect(
                wrapper.containsMatchingElement(<div>{props.traitsCreationLabelText}</div>),
            ).toBeTruthy();
        });
    });

    describe('when user does not have trait creation permission', () => {
        const newProps = {
            ...props,
            canCreateTraits: false,
        };
        const noTraitCreationWrapper = shallow(<SingleSignalTraitsCreation {...newProps} />);

        it('matches the snapshot', () => {
            expect(noTraitCreationWrapper).toMatchSnapshot();
        });

        it('renders "0 Traits"', () => {
            expect(noTraitCreationWrapper.text()).toBe('0 Traits');
        });
    });
});
