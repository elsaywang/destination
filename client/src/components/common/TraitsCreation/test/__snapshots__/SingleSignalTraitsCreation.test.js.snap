// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SingleSignalTraitsCreation/> component when user does not have trait creation permission matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SingleSignalTraitsCreation
    canCreateTraits={false}
    storeSessionAndNavigateToTraits={[Function]}
    traitsCreationLabelText="Create Rule-Based Trait"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): "0 Traits",
  Symbol(enzyme.__nodes__): Array [
    "0 Traits",
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`<SingleSignalTraitsCreation/> component when user has trait creation permission matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SingleSignalTraitsCreation
    canCreateTraits={true}
    storeSessionAndNavigateToTraits={[Function]}
    traitsCreationLabelText="Create Rule-Based Trait"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Add
          size="S"
/>,
        <div
          className="singleCreationlinkText"
>
          Create Rule-Based Trait
</div>,
      ],
      "className": "spectrum-Link singleCreation",
      "data-test": "single-signal-trait-creation",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "size": "S",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Create Rule-Based Trait",
          "className": "singleCreationlinkText",
        },
        "ref": null,
        "rendered": "Create Rule-Based Trait",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Add
            size="S"
/>,
          <div
            className="singleCreationlinkText"
>
            Create Rule-Based Trait
</div>,
        ],
        "className": "spectrum-Link singleCreation",
        "data-test": "single-signal-trait-creation",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "size": "S",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Create Rule-Based Trait",
            "className": "singleCreationlinkText",
          },
          "ref": null,
          "rendered": "Create Rule-Based Trait",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
